JahsieAult/homework.md

```diff
@@ -0,0 +1,164 @@
+>True. You are assigning X to be something and then checking to see if X is equal to it. The answer will always be true. 
+---
+
+All of the following can be solved with ordinary expressions and global variables with primitive values.  You don't need functions, loops, or other topics beyond our first two classes.
+
+ **2)** (_Difficulty: easy_)
+
+Assume variables x, y, and z are numbers.
+
+**a)**
+Write an expression for the mean (i.e. average) of x, y, and z.
+>(x + y + z) / 3
+
+**b)**
+Write a series of expressions to adjust each of x, y, and z halfway toward their mean.
+That is, reset the value of each variable to something based on its previous value.
+
+>var mean = (x + y + z) / 3;
+x = x + (mean/2 - x);
+y = y + (mean/2 - y);
+z = z + (mean/2 -z);
+---
+
+**3)** (_Difficulty: easyish_)
+
+Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+- _l_ : horizontal position of left edge;
+- _r_ : horizontal position of right edge;
+- _t_ : vertical position of top edge;
+- _b_ : vertical position of bottom edge.
+
+**a)**
+Write an expression for the rectangle's area.
+>(Not sure that I understood the placement of the variables) 
+var
+var area = (t - b) * (r - l)
+
+**b)**
+Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+>(t - b) > (r - l)
+
+**c)**
+Write an expression for the circumference of the biggest circle which can fit inside the rectangle.  (Hint: you'll need logic similar to that in **b**.)
+>if ((t - b) > (r - l)); {
+	var maxCircumference = 3.14 * ((r - l)/2); 
+	} else {
+		var maxCircumference = 3.14 * ((t - b)/2);
+	}
+
+**d)**
+Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+
+**e)**
+Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 smaller rectangles, identical in shape but varying by position.
+Define four new variables describing the centermost small rectangle.
+(_Hint: one of the many solutions is very similar to the solution of **2b** above._)
+
+---
+
+**4)** (_Difficulty: easyish_)
+
+Imagine that the squares of an ordinary checkerboard are numbered in two different ways:
+
+* Each square has integer coordinates _(R,C)_ describing its row and column.  Both values should be in the range 0..7, so that the upper-left square is at (0,0) and the bottom-right is at (7,7).
+
+* Each square has a unique integer number N from 0 to 63.  These numbers run sequentially left-to-right one row at a time, top to bottom.  Therefore the upper-left square has N==0 and the bottom-right has N==63.
+
+**a)**  Given a particular R and C, find the corresponding N.  That is, write an expression for variable N containing variables R and C.
+>var N = (R * 8) + (C);
+
+**b)**  Given N, find R.  Write an expression for R which contains N.
+>var R = (N - C)/8;
+Pretty sure I got this one wrong.
```

Comment:
4b) That sounds like it would work to me! You could do it without needing `C` by dividing by 8 and rounding down with `Math.floor`.